app/models/mixins/service_mixin.rb

```diff
@@ -104,23 +108,18 @@ def next_group_index(current_idx, direction = 1)
     nil
   end
 
-  def compact_group_indexes
-    # Remove empty group
-    last_idx = last_group_index
-    return if last_idx == 0
-
-    last_idx.downto(0) do |idx|
-      each_group_resource { |r| r.group_idx -= 1 if r.group_idx >= idx } unless self.group_has_resources?(idx)
+  def circular_reference?(child_svc)
+    return true if child_svc == self
+    if child_svc.kind_of?(Service)
+      ancestor_ids.include?(child_svc.id)
+    elsif child_svc.kind_of?(ServiceTemplate)
+      !!circular_reference_check(child_svc)
```

Comment:
I had deleted everything here. since ancestry takes care of services.
But for service templates, that was not possible. And possibly when it crosses those boundaries from one to the other.